.github/workflows/macos.yml

```diff
@@ -29,7 +29,7 @@ jobs:
     - name: Build
       run: |
         ./autogen.sh
-        ./configure --disable-shared ${{ matrix.configureflags }} CC="ccache clang" CXX="ccache clang++" CXXFLAGS="-O2 -g0 -Wall -Werror -stdlib=libc++ -mmacosx-version-min=10.9 ${{ matrix.cxxflags }}" LDFLAGS="-stdlib=libc++ -mmacosx-version-min=10.9"
+        ./configure --disable-shared ${{ matrix.configureflags }} CC="ccache clang" CXX="ccache clang++" CXXFLAGS="-O2 -g0 -Wall -Werror -stdlib=libc++ -mmacosx-version-min=10.9 ${{ matrix.cxxflags }} --system-header-prefix=boost/" LDFLAGS="-stdlib=libc++ -mmacosx-version-min=10.9"
```

Comment:
I thought that your changes were generating an `-isystem` flag, but I see that's only when using `--with-boost-include`...

Reply:
After digging a bit, it looks like there are several factors at play here:

* The default clang++ on macOS (the one currently used in the CI) is [hardcoded](https://stackoverflow.com/questions/60712261/how-to-prevent-osx-gcc-to-always-search-usr-local-include#comment107445035_60713404) to add `-I /usr/local/include` to the start of the header search path, and I don't think there's any way to add it to the front of the search path with `-isystem` instead. This directory is where homebrew installs the boost headers.
* The [clang++ from homebrew](https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.15-Readme.md#language-and-runtime) located at `$(brew --prefix llvm)/bin/clang` doesn't automatically add `-I /usr/local/include` to the header search path, but unfortunately it doesn't add it as `-isystem /usr/local/include` either.
* If we switch the CI to use the homebrew clang++ we then have two options: a) configure the CI with the `--with-boost-include` flag, or b) just pass `CXXFLAGS="-isystem /usr/local/include"` directly to the CI job. Either way it is similar to how the CMake builds are configured with `"BOOST_ROOT": "/usr"` on Linux, so I don't think it's that bad, and certainly better than `--system-header-prefix`. Do you have any preference between these two options?

Reply:
I think that passing `--with-boost-include` and `--with-boost-lib` to the macos CI is the best option so I've pushed those changes now.

Reply:
The problem is not the CI build — we can pass whatever we need there.  The problem is that on macOS, if people run the default `./configure` they'll get warnings.  Frankly, this time I'd keep the inconvenience of the pragmas in our code to avoid inconvenience to users.

Reply:
Actually if people run the default `./configure` on macOS then they shouldn't see any warnings because `-Wall` isn't added to `CXXFLAGS` by default.

Unless you're suggesting that we want users to see no warnings when running `./configure CXXFLAGS=-Wall` with the default compiler on their platform. In this case it would probably be better to add `-Wall` inside the configure script (and CMakeLists.txt as well) so that it's always included in the default list of compiler options, both in the CI and for all users.

Reply:
You're right, or course, I was thinking of `./configure -Wall` which is not the default but it's not unlikely either.  I wouldn't add warnings by default to a client build (they might have added their code to the build and they're entitled to be relaxed).

However: I've probably figured out how to make this work with autotools.  I'm going to try it and push it to your branch.

But today I've got to do my taxes first 😄

Reply:
Sounds like a plan. Good luck with your taxes!

Reply:
Let's see the CI builds, but it should work now — I did it the autotools way, which is, don't try to detect which compiler you're using or what platform you're on: try to compile with the flag you want and check if you get an error.

To coin a phrase, an elegant weapon for a more civilized age 😄

Reply:
Indeed this is a much more elegant way of configuring the build. Thanks!